internal/api/http/middleware/access_interceptor.go

```diff
@@ -90,6 +96,36 @@ func (a *AccessInterceptor) Handle(next http.Handler) http.Handler {
 	})
 }
 
+func SetExhaustedCookie(cookieHandler *http_utils.CookieHandler, writer http.ResponseWriter, cookieConfig *AccessConfig, instance authz.Instance, requestHost string) {
+	// Limit can only be true when storeOnly is false, so set the cookie and the response code
+	cookieValue, err := templateCookieValue(cookieConfig.ExhaustedCookieValue, instance)
+	if err != nil {
+		// If templating didn't succeed, emit a warning log and just use the plain config
+		logging.WithError(err).WithField("value", cookieConfig.ExhaustedCookieValue).Warning("failed to go template cookie value config")
+	}
+	host, _, err := net.SplitHostPort(requestHost)
+	if err != nil {
+		logging.WithError(err).WithField("host", requestHost).Warning("failed to extract cookie domain from request")
+	}
+	cookieHandler.SetCookie(writer, cookieConfig.ExhaustedCookieKey, host, cookieValue)
+}
```

Comment:
couldn't that be used in `(a *AccessInterceptor) Handle()` function as well?

Reply:
discussed.